# Topic Config
## module: string (sensing/localization/perception/planning/control)
## name: string
## timeout[s]: double (0: none)
## warn_rate[hz]: double (0: none)
topic_configs:
  - module: "sensing"
    name: "/sensing/lidar/pointcloud"
    timeout: 1.0
    warn_rate: 5.0

  - module: "sensing"
    name: "/sensing/lidar/no_ground/pointcloud"
    timeout: 1.0
    warn_rate: 5.0

  - module: "sensing"
    name: "/vehicle/status/twist"
    timeout: 1.0
    warn_rate: 5.0

  - module: "sensing"
    name: "/vehicle/status/steering"
    timeout: 1.0
    warn_rate: 5.0

  - module: "localization"
    name: "/localization/pose_twist_fusion_filter/pose"
    timeout: 1.0
    warn_rate: 5.0

  - module: "perception"
    name: "/perception/object_recognition/detection/objects"
    timeout: 1.0
    warn_rate: 5.0

  - module: "perception"
    name: "/perception/object_recognition/tracking/objects"
    timeout: 1.0
    warn_rate: 5.0

  - module: "perception"
    name: "/perception/object_recognition/objects"
    timeout: 1.0
    warn_rate: 5.0

  - module: "planning"
    name: "/planning/mission_planning/route"
    timeout: 0.0
    warn_rate: 0.0

  - module: "planning"
    name: "/planning/scenario_planning/trajectory"
    timeout: 1.0
    warn_rate: 5.0

  - module: "control"
    name: "/control/vehicle_cmd"
    timeout: 1.0
    warn_rate: 5.0

# Param Config
## name: string
param_configs:
  - module: "sensing"
    name: "/vehicle_info"

# TF Config
## from: string
## to: string
tf_configs:
  - module: "localization"
    from: "map"
    to: "base_link"
    timeout: 1.0

  - module: "localization"
    from: "base_link"
    to: "gnss_link"
    timeout: 1.0

# Blacklist Topics for Simulation Mode
# type: [string]
regex_blacklist_topics:
  - "^/sensing/"
  - "^/localization/pose_twist_fusion_filter/pose$"

# Blacklist Params for Simulation Mode
# type: [string]
regex_blacklist_params: []

# Blacklist TFs for Simulation Mode
# type: [string]
regex_blacklist_tfs:
  - "^gnss_link$"
